doc/imageview_warmups.md

```diff
@@ -0,0 +1,329 @@
+also included the consequences of updating `signal2`: this is because
+the updates happened before the event loop fired again, so by the time
+the `println` statement ran both values had already been updated.
+In some circumstances (like this one), this behavior might be fine or
+even desirable; in other cases, such behavior could be a source of
+bugs.
+
+So in the spirit of exploration, let's look at a second implementation
+that preserves history:
+
+```jl
+module TwoNodes
+
+using Reactive
+
+export Canvas, state1!, state2!
+
+type Canvas{S<:IO}
+    io::S
+    state1::Node{Int}
+    state2::Node{Int}
+
+    function Canvas(io::S, s1::Integer, s2::Integer)
+        n1, n2 = Node(Int, s1), Node(Int, s2)
+        c = new(io, n1, n2)
+        combined = merge(n1, n2)
+        throttled = throttle(1/60, combined)
+        Reactive.preserve(map(x->println(c.io, "state1: $(value(c.state1)); state2: $(value(c.state2))"), throttled))
+        c
+    end
+end
+
+Canvas(io::IO, s1, s2) = Canvas{typeof(io)}(io, s1, s2)
+
+state1!(c::Canvas, val) = push!(c.state1, val)
+state2!(c::Canvas, val) = push!(c.state2, val)
+
+end  # module
+
+using Reactive, TwoNodes
+
+tmr = Timer(_->Reactive.run_till_now(), 0.001, 0.001)
+
+c = Canvas(STDOUT, 1, 1)
+state1!(c, 5)
+sleep(0.1)
+
+state1!(c, 7)
+state2!(c, -3)
+```
+
+Note that the implementation of the `state!` functions was simpler
+here.  With `throttle` we again get the same output:
+
+```jl
+julia> include("twonodes.jl")
+state1: 1; state2: 1
+state1: 5; state2: 1
+
+julia> state1: 7; state2: -3
+julia>
+```
+
+but this time, without `throttle` we get output that respects the history:
+
+```jl
+julia> include("twonodes.jl")
+state1: 1; state2: 1
+state1: 5; state2: 1
+state1: 7; state2: 1
+
+julia> state1: 7; state2: -3
+julia>
+```
+
+## Example 2: sharing state
+
+Suppose we have two `Canvas`es that we want to couple together: for
+example, you might want to show two views of the same image, one in
+"raw" form and the other "annotated" by some kind of image processing
+algorithm.  If you zoom in on one canvas, you might like to
+automatically zoom in on the same region in the other canvas.
+
+```jl
+module TwoSharedNodes
+
+using Reactive
+
+export Canvas, state1!, state2!, relink!
+
+type Canvas{S<:IO}
+    io::S
+    name::ASCIIString
+    state1::Node{Int}
+    state2::Node{Int}
+    update::Node{Void}
+
+    Canvas(io::S, name, s1, s2) = Canvas(io, name, node(Int, s1), node(Int, s2))
+    function Canvas(io::S, name, n1::Node{Int}, n2::Node{Int})
+        c = new(io, name, n1, n2)
+        relink!(c)
+    end
+end
+
+node{T}(::Type{T}, val) = Node(T, val)
+node{T}(::Type{T}, n::Node{T}) = n
+
+function relink!(c::Canvas)
+    isdefined(c, :update) && close(c.update)
+    combined = merge(c.state1, c.state2)
+    throttled = throttle(1/60, combined)
+    c.update = map(x->println(c.io, "$(c.name): state1=$(value(c.state1)), state2=$(value(c.state2))"), throttled)
```

Comment:
I should also say that I may be using Reactive differently than you intended...there clearly are differences in this style of thinking, and I'm only just getting used to it.

But yes, to me this seemed like a case where the `WeakRef` is needed if you want the nodes to be gc()ed. The `close` is presumably sufficient to stop the updates from "rendering," though.